src/builder.rs

```diff
@@ -295,11 +336,16 @@ impl SessionBuilder {
             get_default_control_dir()?
         };
 
+        let prefix = ".ssh-connection";
         let dir = Builder::new()
-            .prefix(".ssh-connection")
+            .prefix(prefix)
             .tempdir_in(socketdir)
             .map_err(Error::Master)?;
 
+        if self.clean_history_control_dir {
+            let _ = clean_history_control_dir(&dir, prefix);
+        }
```

Comment:
You can simplify this by calling `clean_history_control_dir` before creating the new `TempDir`, then you won't have to compare the `entry.path() != dir.path()` and you would also don't need to call `dir.path().parent()` with a `if-let`

Reply:
That's a good suggestion.